Engine/ResourceManagement/Resources/Mesh.cpp

```diff
@@ -98,5 +101,7 @@ void Mesh::LoadInMemory()
 	glEnableVertexAttribArray(7);
 	glVertexAttribPointer(7, 2, GL_FLOAT, GL_FALSE, sizeof(Mesh::Vertex), (void*)offsetof(Mesh::Vertex, tex_coords[1]));
 	glBindVertexArray(0);
+
+	initialized = true;
```

Comment:
Maybe you can move the initialized flag to the Resource.h

Reply:
It is indeed defined on the Resource.h line 47 :D